Create a compatable macro if rpmatch() is not provided by the C library.

This uses an existing configure check.

This is needed for uclibc since it does not have the above function
implemented.

Upstream-Status: Submitted [RPM5 maintainer]

Signed-off-by: Khem Raj <raj.khem@gmail.com>

Updated to rpm 5.4.15+.

Signed-off-by: Mark Hatle <mark.hatle@windriver.com>

Index: rpm/system.h
===================================================================
--- rpm.orig/system.h
+++ rpm/system.h
@@ -358,6 +358,14 @@ extern int _tolower(int) __THROW	/*@*/;
 #include <libgen.h>
 #endif
 
+#ifndef HAVE_RPMATCH
+#define rpmatch(line) \
+	( (line == NULL)? -1 : \
+	  (*line == 'y' || *line == 'Y')? 1 : \
+	  (*line == 'n' || *line == 'N')? 0 : \
+	  -1 )
+#endif
+
 /* -- Retrofit glibc __progname */
 #if defined __GLIBC__ && __GLIBC__ >= 2
 #if __GLIBC_MINOR__ >= 1
